### DISCUSSION TITLE
set up a balance counter

### DISCUSSION THREAD

**Discussion by d-opri** [neuefische/questions#533] [Web]
moin moin,

i am working on a balance counter to display the total of my bills. so far i have a function that calculates the total of all my bill amounts. now i was thinking it would be nice to display the balance for the user, or "you". what i have so far for the general total
```
useEffect(() => {
    if (bill) {
      let balanceTotal = 0;
      bill.forEach((bill) => (balanceTotal += bill.amount));

      setBalance(balanceTotal);
    }
  }, [bill]);
```

 thinking to create a useState yourBalance with the initial value of 0. then in the useEffect would need to filter through the participants and find any with the name value euqal "me, ich" etc., i am doing it in a very clumsy way below..
i would then take the participant.balance and depending on whether its a negative or positive value it adds to or subtracts from yourBalance. i think i am missing a step or two though, any help would be greatly appreciated!
```
function calculateYourTotal((participant, yourBalance) => {
           if ( 
["ich", "Ich", "you", "You", "me", "Me"].includes(participant.name)
)
           {
const balance = Number(participant.balance)
            return balance > 0
              ? yourTotal + balance
              : yourTotal - balance;
          } else {
return yourBalance 
}
    setYourBalance( should be the updated balance but then i am not storing the sum anywhere am i? );
}

```

**Comment by mbosselmann** [comment]
Hi Danie,
I think this is a good idea. For this case a `useEffect` should not be necessary and you can handle the fetched data directly.
I recommend you to work with different array methods to select the balance and to sum it up. For example filter, map and find may be a good solution. 
If you want to sum up the balance then there is one problem: As far as I remember your balance stores a string and not a number. Maybe you should adjust your data structure first to provide a number as balance. 🤔 
I hope my descriptive ideas help you to try out some things. I am looking forward to the current state of your project to look further into your idea.
